src/main/java/com/tabnine/lifecycle/TabNinePluginStateListener.java

```diff
@@ -0,0 +1,31 @@
+package com.tabnine.lifecycle;
+
+import com.intellij.ide.plugins.IdeaPluginDescriptor;
+import com.intellij.ide.plugins.PluginStateListener;
+import com.intellij.openapi.diagnostic.Logger;
+import com.intellij.openapi.extensions.PluginId;
+import com.tabnine.TabNineProcess;
+import com.tabnine.Utils;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.IOException;
+import java.util.Collections;
+
+public class TabNinePluginStateListener implements PluginStateListener {
+
+    @Override
+    public void install(@NotNull IdeaPluginDescriptor descriptor) {
+    }
+
+    @Override
+    public void uninstall(@NotNull IdeaPluginDescriptor descriptor) {
+        final PluginId pluginId = descriptor.getPluginId();
+        if (pluginId != null && pluginId.equals(Utils.getPluginId())) {
+            try {
+                TabNineProcess.startTabNine(true, Collections.singletonList("--uninstalling"));
```

Comment:
just make sure we need --uninstalling and not --uninstalled

Reply:
Yes, that's what we use in sublime plugin